### PR TITLE
Add filtering book page

### PR DESCRIPTION
**Pull request by caesarx26**
# Pull Request Template

## Description

I restyled the category pages to have svg icons for the selected menu and changed the styling of the selected menu that pops up when you long press a category to show up at the end of the list of categories and push the list up instead of the list having a max height that does not cover the menu.

I also implemented the design of the specific book pages. 

I added some test data for the book preview pages shown in the recommended pages and category pages to see how things would look.

 I also fixed the footer and root layout so the footer is always at the bottom of the screen and the specific pages above it take up the whole screen space above the footer. 

Also added support to automaticity make svg icons into components you can make them yourself as components using the syntax for the react native svgs but now you can just import the svg file from its path in the project and use it like it is a component without making the component yourself.
 Could still be good to make the svg component yourself if you want to change more things to it other than height and width.   

## Type of Change

- [ x] Bugfix
- [ x] New Feature
- [ ] Documentation Update
- [ ] Other (please describe):

## Checklist

- [x ] I have read the contributing guidelines.
- [ ] I have followed the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation (if applicable).
- [ x] My changes do not generate new warnings.

## images of how pages and sections look
Screenshot (638)
Screenshot (639)
Screenshot (640)
Screenshot (641)
Screenshot (642)
Screenshot (643)
Screenshot (644)
Screenshot (645)
Screenshot (646)
Screenshot (647)
Screenshot (648)
Screenshot (649)
Screenshot (650)
Screenshot (651)
Screenshot (652)

note no actual data is being stored across the app just test data to see how the data would be displayed and rendered to the user

### CODE REVIEW

**Comment by LilAttacker** [comment]
Of 56 files changed, 28 were .ts or .tsx files.
Of these, just over 1600 lines of code were added.
Of these, only 15 lines were comments.
_You nearly removed as many comments as you added at 12 comments removed._

Now does every little thing need a comment to explain what's going on? Of course not, but when 18 of the 28 files modified have absolutely no comments or explanation whatsoever, then I feel a problem is brewing and its best to fix it now. There are almost no comments in general to explain what is going on or what the purpose of a particular section of code is. 

Now you may understand and know completely what is happening, but does everyone else? Remember, you are not the only one working on this.

If a change was needed to be implemented somewhere in the code could anybody else reliably and quickly accomplish it? Perhaps, but whereas you could potentially begin working on it right away after a brief refresh, anybody else would have to spend time first deciphering the interactions and then working backwards from there to define a purpose to it all before even working on anything.

All of that, when adding a few comments now, when its fresh in your mind on what everything does, could save so much time and heartache later on if we ever decide to go in and change or fix some things. Not to mention given enough time it is unlikely for you to remember what exactly is going on and recall precisely how everything interacted either. If that were to occur, any change we wanted to implement would take so much longer for no good reason.

**Please add comments.**

The comments don't need to be lengthy or extremely descriptive. In fact, its best they aren't. According to chatgpt, a good ratio of comments to aim for is about 10-20% of the code: https://chatgpt.com/share/672079e6-23fc-800f-952c-2b0bc9713fc3. Adding a brief block comment to the top of files and functions to explain their purpose would go a long way to improving the lack of documentation.

--

Next, in TestingMenuSelect.tsx, you have a single comment explaining the code is from chatgpt. I have to ask which code? All of it? Some of it? Which AI model for chatgpt was used?

These things need to be indicated according to the section titled "Instructions for AI-Generated Code in Team Projects" at the bottom under the Projects tab on the class website. It needs to be more substantial than a single line comment according to the assignment.

--

Additionally, you mention aesthetic changes like restyling the category pages in your description. It is quite tough to judge those types of changes without any form of visual aide to use as reference.

Adding a before and after and some basic screenshots with changes highlighted would be useful for us to judge and provide feedback before merging.

--

Finally, I have to ask why is there two sets of reviews (_found in library/[category]/[book].tsx and recommendations/[book].tsx_) and initial books (_found in library/[category]/index.tsx and recommendations/index.tsx_) in the code?

I'm not entirely familiar with type script so perhaps this is asked in ignorance, but do these need to be separate like this? Can they not be condensed down into a single file? Or is this only temporary to show proof of concept and isn't actually how this is going to be implemented?

--

Notably, there are no tests for any of this. But we did assign someone to fix that, so maybe this will be covered later? If others do approve of this, then I'm willing to overlook it. But we do need to start adding tests to PRs before merging.

I also have a sperate issue with how sparse your description of this PR is given its size. Over 1600 lines added, 56 files changed, and you have one run-on sentence to describe it all? A more thorough description or a smaller PR would be much appreciated and make this much easier to review. Smaller and more frequent PRs would also mean less of a chance for a merge conflict to occur and give everyone a more similar view of what the current state of the project is.

File: Tabby/components/categories/TestingMenuSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// all code from chat gpt
```
Comment (LilAttacker) [inline Tabby/components/categories/TestingMenuSelect.tsx:1]: This needs more of a description according to the project requirements on the class website: https://johnxu21.github.io/teaching/CS472/project/

Comment (caesarx26) [reply Tabby/components/categories/TestingMenuSelect.tsx:1]: I deleted it it is not being used I was trying to see if I could get something to work with a drag and drop menu without using a package but will probably use something like: https://www.npmjs.com/package/react-native-draggable-flatlist?activeTab=readme

Comment (caesarx26) [reply Tabby/components/categories/TestingMenuSelect.tsx]: for the two sets of reviews this is just test data for now right now there is no actual book data being stored just components or sections of pages that take the test data and display it and do actions based on it to just have how the data should be displayed and looked

Comment (caesarx26) [reply Tabby/components/categories/TestingMenuSelect.tsx]: Also added some more comments to the code

**Comment by caesarx26** [comment]
for the two sets of reviews this is just test data for now right now there is no actual book data being stored just components or sections of pages that take the test data and display it and do actions based on it to just have how the data should be displayed and looked

**Comment by caesarx26** [comment]
I also added some more comments to the code.

**Comment by caesarx26** [comment]
The pull request is big I deleted a lot of old things as well that I had from the sample project from free code camp. And in the future will have smaller pull requests and split stuff up.

**Comment by caesarx26** [comment]
tests will be added later by David will mostly test functions used within components like the sorting categories function and other functions used in the category page or in the stars Rating components for now the actual ui interactions will add testing for that later.

**Comment by caesarx26** [comment]
> I also added some more comments to the code.

There might not be a lot of comments on specific components or sections of pages if all they are doing or mostly doing is just returning tsx and they do not have a lot of functions or dynamic behavior within that component/section. By looking at the tsx you can see the styles being applied or props used to set what the component will do. 

For like the category pages and components used there like the rename modal there is a lot of comments on that page and component since there is a lot of dynamic behavior and functions passed from the category page to its modals. But things like the custom icon components I made, usually I just pass a flag or specific color to change the color of it or height and width. 

- Or for example the book card component that gets data from a book and just displays it based on the book passed 
- there is not going to be a lot of comments for this since it is just displaying the data passed and you can see what components are being used and what styles are being applied directly 
```TypeScript
    const BookCard: React.FC<BookCardProps> = ({ book }) => {

    return (
        <View className="pl-7 pt-7">
            <View className="flex-row">
                <Image
                    source={{ uri: book.image }}
                    className="w-28 h-40"
                />

                <View className="flex-col pl-4">
                    <Text className="text-lg font-bold text-white">{book.title}</Text>
                    <Text className="text-md text-white font-semibold italic">{book.author}</Text>
                    <Text className="mt-3 text-lg text-white">Rating</Text>
                    <StarsRating rating={book.rating || 3} />
                </View>
            </View>

            <View className="pt-2">
                <Text className="text-lg text-white">Summary</Text>
                <ScrollView className='max-h-40'>
                    <Text className="text-sm text-white">{book.summary}</Text>
                </ScrollView>

            </View>
        </View>



    );

};
```

**Comment by caesarx26** [comment]
Updated pull request to have more pictures and to be more descriptive

**Review by LilAttacker — APPROVED** [review]
After looking it over, I approve.

Thank you for addressing each of my issues and resolving them. It looks good and I have no more concerns to raise.

Although Josh and David should still have their own reviews and approvals of this since they are the ones far more likely to be working on this.

**Comment by MrSnoodle01** [comment]
Everything looks good. I think that the amount of comments that you have is fine, the vast majority of the html that you added is self-explanatory and doesn't need comments. And it looks like you added comments where they were needed most. One thing to think about in the future is that this is quite a large pull request that could have been broken up into smaller chunks. Having smaller more manageable pull requests will make reviewing easier and allow the team to see what is getting done earlier.

**Review by MrSnoodle01 — APPROVED** [review]

**Review by David1000734 — APPROVED** [review]
Looks really good, nothing to really say.

However, I noticed that a lot of the colors are still using the hex value instead of colors we added to the tailwind.config.js file. For the next pull request, can we try our best to try to adhere to that practice? I'm guilty of this too and will be reminding myself to do this for my future commits.

**Pull request merged**